### PR TITLE
acceptance test for checking shares pointing to deleted groups are not retrieved 

### PR DESCRIPTION
**Pull request by haribhandari07**
## Description

Includes api and webui tests to verifiy shares that point to deleted groups are not retrieved.
## Related Issue

https://github.com/owncloud/QA/issues/357
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/apiShareManagementBasic/createShare.feature [hunk 1]
```diff
@@ -1130,3 +1130,22 @@ Feature: sharing
       | ocs_api_version | ocs_status_code | http_status_code |
       | 1               | 100             | 200              |
       | 2               | 200             | 200              |
+
+   Scenario Outline: shares shared to deleted group should not be available
+     Given using OCS API version "<ocs_api_version>"
+     And parameter "shareapi_only_share_with_membership_groups" of app "core" has been set to "yes"
```
Comment (dpakach) [inline tests/acceptance/features/apiShareManagementBasic/createShare.feature]: Is this step required here?

File: tests/acceptance/features/apiShareManagementBasic/createShare.feature [hunk 2]
```diff
@@ -1130,3 +1130,22 @@ Feature: sharing
...
+       | username |
+       | user1    |
+       | user2    |
+     And group "grp1" has been created
+     And user "user1" has been added to group "grp1"
+     And user "user2" has been added to group "grp1"
+     When user "user1" shares file "/textfile0.txt" with group "grp1" using the sharing API
+     Then as "user2" file "/textfile0 (2).txt" should exist
+     And group "grp1" has been deleted
+     Then as "user2" file "/textfile0 (2).txt" should not exist
```
Comment (dpakach) [inline tests/acceptance/features/apiShareManagementBasic/createShare.feature]:
```suggestion
     And as "user2" file "/textfile0 (2).txt" should not exist
```

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature [hunk 3]
```diff
@@ -299,3 +299,19 @@ Feature: Sharing files and folders with internal groups
     When user "user1" logs in using the webUI
     Then folder "test" should be listed on the webUI
     And folder "test (2)" should be listed on the webUI
+
+  Scenario: shares shared to deleted group should not be available
+    Given user "user1" has uploaded file "filesForUpload/lorem.txt" to "/lorem.txt"
+    And user "user1" has logged in using the webUI
+    And the user has shared folder "/lorem.txt" with group "grp1"
+    And the user re-logs in as "user2" using the webUI
+    Then file "lorem.txt" should be marked as shared with "grp1" by "User One" on the webUI
```
Comment (dpakach) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature]: I dont think this step is required here.

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature [hunk 4]
```diff
@@ -299,3 +299,19 @@ Feature: Sharing files and folders with internal groups
...
+    And group "grp1" has been deleted
+    When the user reloads the current page of the webUI
+    Then file "lorem.txt" should not be listed on the webUI
+    When the user browses to the shared-with-you page
+    Then file "lorem.txt" should not be listed on the webUI
+    And as "user2" file "/lorem.txt" should not exist
+    And the user re-logs in as "user1" using the webUI
+    And the user opens the share dialog for file "lorem.txt"
+    Then the user "grp1" should not be in share with user list
```
Comment (dpakach) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature]:
```suggestion
    And the user "grp1" should not be in share with user list
```

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature [hunk 5]
```diff
@@ -299,3 +299,19 @@ Feature: Sharing files and folders with internal groups
...
+    And the user has shared file "/lorem.txt" with group "grp1"
+    And the user re-logs in as "user2" using the webUI
+    Then file "lorem.txt" should be marked as shared with "grp1" by "User One" on the webUI
+    When the administrator deletes group "grp1" using the provisioning API
+    And the user reloads the current page of the webUI
+    Then file "lorem.txt" should not be listed on the webUI
+    When the user browses to the shared-with-you page
+    Then file "lorem.txt" should not be listed on the webUI
+    And as "user2" file "/lorem.txt" should not exist
+    And the user re-logs in as "user1" using the webUI
```
Comment (dpakach) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature]:
```suggestion
    When the user re-logs in as "user1" using the webUI
```

File: tests/acceptance/features/apiShareManagementBasic/createShare.feature [hunk 6]
```diff
@@ -1130,3 +1130,21 @@ Feature: sharing
...
+     And these users have been created with default attributes and skeleton files:
+       | username |
+       | user1    |
+       | user2    |
+     And group "grp1" has been created
+     And user "user1" has been added to group "grp1"
+     And user "user2" has been added to group "grp1"
+     When user "user1" shares file "/textfile0.txt" with group "grp1" using the sharing API
+     Then as "user2" file "/textfile0 (2).txt" should exist
+     And group "grp1" has been deleted
```
Comment (dpakach) [inline tests/acceptance/features/apiShareManagementBasic/createShare.feature]: This is a given step, I think a When step should be used here.

File: tests/acceptance/features/apiShareManagementBasic/createShare.feature [hunk 7]
```diff
@@ -1130,3 +1130,21 @@ Feature: sharing
       | ocs_api_version | ocs_status_code | http_status_code |
       | 1               | 100             | 200              |
       | 2               | 200             | 200              |
+
+   Scenario Outline: shares shared to deleted group should not be available
+     Given using OCS API version "<ocs_api_version>"
+     And these users have been created with default attributes and skeleton files:
+       | username |
+       | user1    |
+       | user2    |
```
Comment (skshetry) [inline tests/acceptance/features/apiShareManagementBasic/createShare.feature]: Share using `user0` to the group and check for both users.

File: tests/acceptance/features/lib/FilesPageElement/SharingDialog.php [hunk 8]
```diff
@@ -649,7 +649,24 @@ public function checkPublicLinkCount(Session $session, $count) {
 	public function isUserPresentInShareWithList($username) {
 		$shareWithList = $this->getShareWithList();
 		foreach ($shareWithList as $userOrGroup) {
-			if ($userOrGroup->find('xpath', $this->userNameSpanXpath)->getHtml() === $username) {
+			if ($userOrGroup->find('xpath', $this->userOrGroupNameSpanXpath)->getHtml() === $username) {
```
Comment (skshetry) [inline tests/acceptance/features/lib/FilesPageElement/SharingDialog.php]: Please check if the element is null or not.

File: tests/acceptance/features/lib/FilesPageElement/SharingDialog.php [hunk 9]
```diff
@@ -649,7 +649,24 @@ public function checkPublicLinkCount(Session $session, $count) {
...
+	 * check if group with the given groupName is present in the shared with list
+	 *
+	 * @param string $groupName
+	 *
+	 * @return bool
+	 */
+	public function isGroupPresentInShareWithList($groupName) {
+		$shareWithList = $this->getShareWithList();
+		foreach ($shareWithList as $group) {
+			if ($group->find('xpath', $this->userOrGroupNameSpanXpath)->getHtml() === $groupName . " (group)") {
```
Comment (skshetry) [inline tests/acceptance/features/lib/FilesPageElement/SharingDialog.php]: same here

File: tests/acceptance/features/lib/FilesPageElement/SharingDialog.php [hunk 10]
```diff
@@ -648,8 +648,39 @@ public function checkPublicLinkCount(Session $session, $count) {
...
-			if ($userOrGroup->find('xpath', $this->userNameSpanXpath)->getHtml() === $username) {
+		foreach ($shareWithList as $user) {
+			$expectedUsername = $user->find('xpath', $this->userOrGroupNameSpanXpath);
+			$this->assertElementNotNull(
+				$expectedUsername,
+				__METHOD__ .
+				" xpath $this->userOrGroupNameSpanXpath " .
+				"could not find user in share with list "
+			);
+			if ($user->find('xpath', $this->userOrGroupNameSpanXpath)->getText() === $username) {
```
Comment (paurakhsharma) [inline tests/acceptance/features/lib/FilesPageElement/SharingDialog.php]:
```suggestion
			if ($expectedUsername->getText() === $username) {
```

File: tests/acceptance/features/lib/FilesPageElement/SharingDialog.php [hunk 11]
```diff
@@ -648,8 +648,39 @@ public function checkPublicLinkCount(Session $session, $count) {
...
+			$this->assertElementNotNull(
+				$expectedGroupName,
+				__METHOD__ .
+				" xpath $this->userOrGroupNameSpanXpath " .
+				"could not find group in share with list "
+			);
+			if ($expectedGroupName->getText() === $groupName . " (group)") {
 				return true;
 			}
 		}
```
Comment (paurakhsharma) [inline tests/acceptance/features/lib/FilesPageElement/SharingDialog.php]: please return false here.

**Review by paurakhsharma — COMMENTED** [review]
Just some minor comments, please have a look.

File: tests/acceptance/features/lib/FilesPageElement/SharingDialog.php [hunk 12]
```diff
@@ -648,8 +648,39 @@ public function checkPublicLinkCount(Session $session, $count) {
...
-			if ($userOrGroup->find('xpath', $this->userNameSpanXpath)->getHtml() === $username) {
+		foreach ($shareWithList as $user) {
+			$actualUsername = $user->find('xpath', $this->userOrGroupNameSpanXpath);
+			$this->assertElementNotNull(
+				$actualUsername,
+				__METHOD__ .
+				" xpath $this->userOrGroupNameSpanXpath " .
+				"could not find user in share with list "
+			);
+			if ($actualUsername->getText() === $username) {
```
Comment (individual-it) [inline tests/acceptance/features/lib/FilesPageElement/SharingDialog.php]: why changing from `getHtml`? If its needed please confirm it has no side effects and works corrects

**Review by individual-it — CHANGES_REQUESTED** [review]
to fix https://github.com/owncloud/QA/issues/357
we need more checks. We need to check not only if the share is visible at the receivers end but also what the underlying share informations are.
you can do that by sending `GET` to ocs api sharing endpoint `/apps/files_sharing/api/v1/shares`
see https://github.com/owncloud/core/blob/e17ced4cb34bf1cff90a454b677ffdd441d4b59a/tests/acceptance/features/apiShareManagementBasic/createShare.feature#L542

**Review by individual-it — APPROVED** [review]

**Comment by HariBhandari07** [comment]
backport on https://github.com/owncloud/core/pull/35852

**Comment by phil-davis** [comment]
restarted drone - master CI should be OK now.

**Pull request merged**